Tasks/Common/coveragepublisher/coveragepublisher.ts

```diff
@@ -74,7 +74,23 @@ async function publishCoverage(inputFiles: string[], reportDirectory: string, pa
             "PIPELINES_COVERAGEPUBLISHER_DEBUG": taskLib.getVariable('PIPELINES_COVERAGEPUBLISHER_DEBUG'),
             "HTTPS_PROXY": process.env['HTTPS_PROXY'],
             "NO_PROXY": process.env['NO_PROXY'],
-            "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT": taskLib.getVariable('DOTNET_SYSTEM_GLOBALIZATION_INVARIANT')
+            "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT": taskLib.getVariable('DOTNET_SYSTEM_GLOBALIZATION_INVARIANT'),
+            // NTLM proxy support
+            "NTLM_PROXY": process.env['NTLM_PROXY'] || taskLib.getVariable('NTLM_PROXY'),
+            "NTLM_USERNAME": process.env['NTLM_USERNAME'] || taskLib.getVariable('NTLM_USERNAME'),
+            "NTLM_PASSWORD": process.env['NTLM_PASSWORD'] || taskLib.getVariable('NTLM_PASSWORD'),
```

Comment:
NTLM_PASSWORD may be supplied via environment variables (process.env). When values are pulled from process.env, they aren’t automatically masked by the Azure Pipelines agent, so the password could leak into logs. It would be safer to always use taskLib.getVariable() for these secrets and to call taskLib.setSecret() so they’re obfuscated in logs.

Reply:
Actually here if we are offering a way to enter the ntlm password then yah no choice, we can't avoid it, so ignore my original comment